service/controller/resource/servicemonitor/service/cluster_autoscaler.go

```diff
@@ -0,0 +1,91 @@
+package service
```

Comment:
It'd be better to have this in a separate PR as this one is quite big and appears to cover a number of things making it a bit hard to review. See also [our docs on PRs](https://intranet.giantswarm.io/docs/dev-and-releng/developer-workflow/pull-requests/). It's more of a note for the future, probably not worth re-working this one. Not sure how inter-dependent those changes are.

Reply:
I agree, and I did not plan on making it that big